### DISCUSSION TITLE
Recording on external storage

### DISCUSSION THREAD

**Discussion by ghost** [blakeblackshear/frigate#3802] [Ask A Question]
Hi
Does anyone know a simple way to record video streams directly on an external storage like a hard drive / usb flash drive ?
Thanks

**Comment by NickM-27** [comment]
Depends how you're running, but if you're running the recommended way with docker in Linux then you'd just mount whatever storage drive you have and then use that as the volume passed to frigate

https://docs.frigate.video/installation#storage

**Reply by ghost** [reply]
Thanks for your help, I obviously read the doc too quickly

I realized I was running frigate as an addon in HA, so I tried to install it by the recommended way.
Since I was not able to figure out how to use docker compose in HassOS, I tried the `docker run` way : 
```
docker run -d \
  --name frigate \
  --restart=unless-stopped \
  --mount type=tmpfs,target=/tmp/cache,tmpfs-size=1000000000 \
  --device /dev/bus/usb:/dev/bus/usb \
  --shm-size=128m \
  -v /home/ha/frigate/media:/media/frigate \
  -v /home/ha/frigate/config.yml:/config/config.yml:ro \
  -v /etc/localtime:/etc/localtime:ro \
  -e FRIGATE_RTSP_PASSWORD='password' \
  -p 5000:5000 \
  -p 1935:1935 \
  blakeblackshear/frigate:stable-aarch64
```
([https://docs.frigate.video/installation/#docker](https://docs.frigate.video/installation/#docker)), but I got the error: "docker: Error response from daemon: error while creating mount source path '/home/ha/frigate/config.yml': mkdir /home/ha: read-only file system"
I recursively modified permissions to 777 but nothing changed.
I have no idea how to fix this

If you know how to use docker compose in HassOS or how to fix the error, please let me know

**Reply by NickM-27** [reply]
HassOS does not support running the containers that way. It needs to be run as an addon in HassOS or run in a separate OS in docker.

**Reply by demusss** [reply]
I'm currently running HassOS and struggling getting an external drive connected for Frigate.

Are you saying its best to have a debian system with Home Assistant Container (Docker) and then Frigate installed also as Docker ?

**Reply by NickM-27** [reply]
@demusss I am saying that the best way to accomplish that is to run Frigate installed in native docker on a debian (or other linux) system.

How HA is installed does not directly matter. I personally use HA installed as the core docker container, but frigate installed in native docker can work with either.

**Comment by Cold-Lemonade** [comment]
I hate to hijack someone else's thread, but I have a related question. I am running Frigate in docker and do not have the issue of the OP. Instead, my question is whether /media/frigate and /config/config.yml:ro should point to the same physical device.

For example, I have a 6TB hard drive in a USB 3.0 enclosure. My plan is to have this device be /media/frigate. Is it important for /config/config.yml:ro to be on the same device?

Relatedly, where on the computer will the Frigate files/programs be saved? I understand that they run in a docker container, but which physical device will they be on?

Thanks.

**Reply by NickM-27** [reply]
> For example, I have a 6TB hard drive in a USB 3.0 enclosure. My plan is to have this device be /media/frigate. Is it important for /config/config.yml:ro to be on the same device?

There is no reason they need to be on the same physical device. 

> 
> Relatedly, where on the computer will the Frigate files/programs be saved? I understand that they run in a docker container, but which physical device will they be on?

They will be on whichever physical device you put them on based on your volume mapping

**Reply by mdisabato** [reply]
In my case, I have everything but the database on the NAS. This was intentional to allow config.yml to be visible without going into the RPi running Frigate. I use Portainer to stop Frigate. Modify config.yml. Lastly, restart in Portainer. That has been less of an issue since I moved to RC2. I'm using the built in editor for minor edits. However, I still modify config.yml when I need things like find/replace. This also allows the nightly backups to pick up the Frigate folders.

**Reply by ronniekw** [reply]
Thank you all for god ideas and inputs. 

I think I found an error in my directory name in docker-compose.yml.

Now it looks better:
![image](https://github.com/blakeblackshear/frigate/assets/52671597/9cfea2d3-7423-43b8-9bb6-0b7227ae59ec)

Now I will wait and see if the recordings are being saved.

Thanks for fast trouble shooting :)

Regards
Ronnie
